### PR TITLE
Makes AI camera track link not break on weird characters

### LINKED ISSUE #9666: Names with apostrophes break AI tracking links

### PR DESCRIPTION
**Pull request by maartenberg**
I tested this using a local server with an AI with an apostrophe in the name, it works.
Fixes #9666.

### CODE REVIEW

**Comment by spasticVerbalizer** [comment]
Oops, the tcs-changed names still break

**Comment by Aranclanos** [comment]
this should be merged anyways

**Comment by Miauw62** [comment]
Yeah this is still a good PR, the tcs thing is a separate issue really
Op 28-mei-2015 22:40 schreef "spasticVerbalizer" <notifications@github.com>:

**Comment by Aranclanos** [comment]
cover all href links that use AI tracking

**Comment by spasticVerbalizer** [comment]
I did a search for "track=" and I think I've got all of the hrefs, and the tcs issue seems to be resolved with M.name, but I don't know how this works with voice-changers etc. as I haven't been able to test that.

File: code/modules/mob/living/silicon/ai/say.dm [hunk 1]
```diff
@@ -7,7 +7,7 @@
 /mob/living/silicon/ai/compose_track_href(atom/movable/speaker, namepart)
 	var/mob/M = speaker.GetSource()
 	if(M)
-		return "<a href='?src=\ref[src];track=[namepart]'>"
+		return "<a href='?src=\ref[src];track=[html_encode(M.name)]'>"
```
Comment (Aranclanos) [inline code/modules/mob/living/silicon/ai/say.dm]: don't switch to M.name, people will show up as 'unknown' in the radio without a voice changing mask. Use the 'namepart' argument

**Comment by Aranclanos** [comment]
you got them all, just change that last thingie and this is good to go

**Comment by Aranclanos** [comment]
👌

**Pull request merged**